src/Features/Core/Portable/FindUsages/DefinitionItem.cs

```diff
@@ -254,6 +257,13 @@ public virtual Task<bool> TryNavigateToAsync(Workspace workspace, bool showInPre
                                    .Add(MetadataSymbolOriginatingProjectIdGuid, projectId.Id.ToString())
                                    .Add(MetadataSymbolOriginatingProjectIdDebugName, projectId.DebugName ?? "");
 
+            // Find the highest level containing type to show as the "file name". For metadata locations
+            // that come from embedded source or SourceLink this will be wrong, but its _way_ too expensive
```

Comment:
can you clarify how it will be wrong?

Reply:
```suggestion
            // that come from embedded source or SourceLink this could be wrong, as there is no reason
            // to assume a type is defined in a filename that matches, but its _way_ too expensive
```